userspace/libsinsp/sinsp.cpp

```diff
@@ -1297,8 +1299,6 @@ int32_t sinsp::next(OUT sinsp_evt **puevt)
 				{
 					m_external_event_processor->process_event(NULL, libsinsp::EVENT_RETURN_TIMEOUT);
 				}
-				*puevt = NULL;
-				return res;
```

Comment:
Avoid multiple `return`s here; just use the return at L1339 that was already there.